rse-py-scripting.Rmd

```diff
@@ -103,42 +133,461 @@ optional arguments:
   -h, --help  show this help message and exit
 ```
 
-and issues errors when users give the program invalid arguments:
+As the `--help` hint shows,
+`argparse` can handle [long options][long-option-shell] as well as traditional single-letter options;
+the online help has examples of this and much more.
+
+## How can I configure common options for a program? {#rse-py-scripting-configure}
+
+A program that does exactly the same thing every time we run it
+isn't as useful as one that can work on different files
+or analyze data with different thresholds.
+Software of all kinds needs to be controlled;
+some things change more often than others,
+so we need a simple, uniform way to specify some options and leave others alone.
+
+The modern Unix convention is to provide multiple levels of configuration:
+
+1.  A system-wide configuration file for general settings.
+2.  A user-specific configuration file for personal preferences.
+3.  A job-specific file with settings for a specific run.
+4.  Command-line options to change things that commonly change.
+
+This is sometimes called [overlay configuration][overlay-configuration]
+because each level overrides the ones above it:
+the user's configuration file overrides the system settings,
+the job configuration overrides the user's defaults,
+and the command-line options overrides that.
+
+Programmers have invented many formats for configuration files,
+so please do not create your own.
```

Comment:
Do we need list/discuss all the different options for configuration files, or should we just cut to the chase and say that YAML is popular so we're going with that?

Reply:
I included it because people are still likely to run into it. I took out the mention of Python data structures as a configuration format.